inc/saiswitch.h

```diff
@@ -2106,6 +2106,22 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_MACSEC_OBJECT_ID,
 
+    /**
+     * @brief Next hop group member state change notification callback
+     * function passed to the adapter.
+     *
+     * In case driver does not support this attribute, The Host adapter should poll
+     * port status by SAI_NEXT_HOP_GROUP_MEMBER_ATTR_OPER_STATUS.
+     *
+     * Use sai_next_hop_group_member_state_change_notification_fn
+     * as notification function.
+     *
+     * @type sai_pointer_t sai_next_hop_group_member_state_change_notification_fn
+     * @flags CREATE_AND_SET
+     * @default NULL
+     */
+    SAI_SWITCH_ATTR_NEXT_HOP_MEMBER_OPER_STATE_CHANGE_NOTIFY,
+
```

Comment:
Just for completeness - should we also add SAI_SWITCH_ATTR_NEXT_HOP_GROUP_OPER_STATE_CHANGE_NOTIFY? This could help with the case when all members of the group are down - especially, when detected through means other than control plane marking them as down?

Reply:
Do you have something particular in mind? Such notification should be well defined.

Reply:
When all the nexthops of a nexthop group go into SAI_NEXT_HOP_GROUP_MEMBER_OPER_STATUS_INACTIVE - which generally means no traffic can be forwarded through this nexthop group, it is good to get a notification into the application space of such an event. In some NOS's today, control plane detects this  and notify that a route is not reachable due to this reason. Also, there are some dataplanes (at least one) that actually fast detect port/neighbor failures to accumulate and allow for a notification of the group has all its members down. This is typically a faster notification to the application - routing, for instance, to recompute/resolve to an alternate path/nexthop. BFD based methods of nexthop liveliness are used for these. We don;t have to get it in this release if you think it is not very interesting for your proposal. Thanks

Reply:
This can be indeed helpful. But it comes down to counting active members and so can also be done in the application